### DISCUSSION TITLE
Discussion about Stack tabs support

### DISCUSSION THREAD

**Discussion by rohanbatraio** [RafaelGB/obsidian-db-folder#472] [General]
I literally after trying to find problem for hours on #469 . I give up, but with ongoing debugging here's what i understand that would just be a theory for now, that's why opened a discussions rather then replying on the initial bug report, as really these are just my speculations. 

1. The react table rendering is not optimized while running an intense query in combination be it tags, dataview or whatever data as a source to dbfolder  {especially in sliding panes plugin, which have this on their [readme](https://github.com/deathau/sliding-panes-obsidian#compatibility)  "The current API of this repo targets Obsidian v0.10.9.")}. That's a bit misleading because the plugin last updated on September 2022. I am confused about sliding pane, whether on what api version it is targeting. As from #469 many others have indicated that they are having problems with sliding panes as the root cause, and I myself was one of them. But when tried with a mock vault things tend to look different.
like for example:  I ran a mock vault, everything was fine with sliding panes. Perhaps having one or two files on two databases works just fine, but when your are opening that in concurrent with large query then things lead to a crash.

2. What i am about to write, is happening right at this moment in my obsidian, most of the time you resize the database pane by whatever means, like for example: 
* Opening navigation window
* Opening right sidebar etc. etc.

then obsidian is struck, and this happened right at this moment i am writing, i just opened my calender plugin  pane, which i am sure most of you have been using, and yeah obsidian is struck. (calender plugin has nothing to do with the issue, but resizing of tables does.) 

Note -> You would definitely don't see any of this when using 5 to 6 files with a single db and opening them concurrently, that time things are as smooth as butter, But combine pages with n+ rows in a database, querying and opening multiple dbs of the same nature, would end up in these issues. 

Conclusion (on basis of speculations ofcourse) :- rendering engine needs to scale and load handling should be done in a more crash resilient way. 

that's all for hours of debugging. :laughing: need some more hours to come back and write what happened. 

in a few days, i am planning to release my vault of academia as opensource, then you can replicate these issues @RafaelGB. But ofcourse cleaning my vault, preparing it for public would take time. You know what it's like, right.

**Comment by rohanbatraio** [comment]
Failure -> Can't reproduce any of this on a mock vault. Probably because of point ^1 as the mock vault doesn't have enough query and load. Will try again.

**Reply by RafaelGB** [reply]
great investigative work!

With insider version I could see that Sliding panes is deprecated and will be removed in favor of stack tabs feature, so I will focus on https://github.com/RafaelGB/obsidian-db-folder/issues/435

While its fixed, I just add some error handler with the search, but I think its not related. It has to do with refreshing the table after filtering. With the new stack tabs it doesn't even load well

**Reply by RafaelGB** [reply]
Okay, the problem its related with the overflow css property. Table and stack shares the same overflow and this cause the crash...


https://user-images.githubusercontent.com/11924043/194394714-3968d47f-37b5-437d-b9e3-2dc7e1f46734.mov


The tables need its own overflow property in case of belong to one. But this fact requires some refactor. Once this theory is confirmed I will let you know

**Comment by RafaelGB** [comment]
Confirmed!

https://user-images.githubusercontent.com/11924043/194400165-63f10e95-3f38-4a4a-a8ef-d0d48efd330b.mov

This is a css job (not my special strong point), so do not expect a fix soon =(

**Reply by rohanbatraio** [reply]
At last we know what the problem is, that's all what matters for now :smile:
